google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -1110,6 +1150,7 @@ def _on_response(self, response: gapic_types.StreamingPullResponse) -> None:
                 )
                 return
 
+            i: int = 0
             for received_message in received_messages:
```

Comment:
Not a python expert, but does using enumerate work better here instead of incrementing the counter manually?

Reply:
Reason for not trying to enumerate here is that `subscribe_opentelemetry[]` can either be having the same length as the received_messages list when opentelemetry is enabled, in which case an enumeration can be used, but `subscribe_opentelemetry` is an empty list when opentelemetry is disabled. In this case, enumeration would not work, because received_messages and subscribe_opentelemetry would have N and zero length, and something along the lines of:
`for x, y in zip(received_messages, subscribe_opentelemetry)` would cause issues. Hence, I used just a normal counter here.